project-base/storefront/components/Blocks/Adverts/Adverts.tsx

```diff
@@ -33,57 +30,25 @@ export const Adverts: FC<AdvertsProps> = ({
     className,
     isSingle,
 }) => {
-    const [{ data: advertsData }] = useAdvertsQuery();
-
-    const filteredAdverts = advertsData?.adverts.filter((advert) => advert.positionName === positionName);
+    const [{ data: advertsData }] = useAdvertsQuery({
+        variables: {
+            categoryUuid: currentCategory?.uuid || null,
+            positionNames: [positionName],
+        },
+    });
```

Comment:
I have noticed that you call this query with multiple positions (e.g. in `getServerSideProps` in `[...categorySlug].tsx`) and then call it in components with a single position. This way the SSR cache won't work. You have to call the query with the same variables both on the server and on the client in order to make it work.

https://github.com/user-attachments/assets/8e686e92-0375-4453-80dc-df9d6ecba039